app/controllers/user.js

```diff
@@ -16,6 +16,58 @@ const passwordValid = password => {
   return isAlphanumeric.test(password) && password.length >= 8;
 };
 
+exports.adminSignUp = (req, res, next) => {
+  User.getUserByEmail(req.body.email).then(existingUser => {
+    if (existingUser) {
+      bcrypt.compare(req.body.password, existingUser.password).then(isValid => {
+        if (isValid) {
+          User.update({ admin: true }, { returning: true, where: { email: req.body.email } })
+            .then(() => {
+              res.status(200);
+              res.send({ newAdmin: existingUser });
+            })
+            .catch(err => {
+              next(err);
+            });
+        } else {
+          return next(errors.incorrectCredentials);
+        }
+      });
+    } else {
+      if (!emailValid(req.body.email)) {
```

Comment:
we are having a lot of repeated code between this function and the normal user creation, maybe you can find a way to abstract it and re use it in both places.